resources/js/components/ui/Modal/Modal.tsx

```diff
@@ -0,0 +1,53 @@
+import * as dialog from "@zag-js/dialog";
+import {normalizeProps, useMachine} from "@zag-js/solid";
+import {ParentComponent, Show, createComputed, createMemo, createUniqueId} from "solid-js";
+import {Portal} from "solid-js/web";
+import s from "./Modal.module.scss";
+
+interface Props {
+  title?: string;
+  open: boolean;
+  /**
+   * Event triggered when the Esc key is pressed or when the user clicks outside of the modal.
+   * If specified, it can react to that event by closing the dialog (i.e. setting the open prop
+   * to false).
+   */
+  onEscape?: (reason: "key" | "click") => void;
+}
+
+export const Modal: ParentComponent<Props> = (props) => {
+  const [state, send] = useMachine(
+    dialog.machine({
+      closeOnOutsideClick: false,
+      closeOnEsc: false,
+      onEsc: () => props.onEscape?.("key"),
+      onOutsideClick: () => props.onEscape?.("click"),
```

Comment:
skoro wyłączasz closeOnEsc, to ta opcja z key zadziała?

Reply:
Wyłączam automatyczne ukrywanie na Esc ale nadal mogę obsłużyć Esc handlerem. To mi się podoba w tym zag'u że do pewnych mechanizmów daje dostęp do więcej niż jednej warstwy gdzie można je obsłużyć.